solver/src/naive_solver.rs

```diff
@@ -0,0 +1,144 @@
+mod two_order_settlement;
+
+use crate::settlement::Settlement;
+use model::{OrderKind, TokenPair, UserOrder};
+use primitive_types::U512;
+use std::{cmp::Ordering, collections::HashMap};
+
+pub fn settle(orders: impl Iterator<Item = UserOrder>) -> Option<Settlement> {
+    let orders = organize_orders_by_token_pair(orders);
+    // TODO: Settle multiple token pairs in one settlement.
+    orders
+        .into_iter()
+        .find_map(|(_, orders)| settle_pair(orders))
+}
+
+fn settle_pair(orders: TokenPairOrders) -> Option<Settlement> {
+    let most_lenient_a = orders.sell_token_0.into_iter().min_by(order_by_price)?;
+    let most_lenient_b = orders.sell_token_1.into_iter().min_by(order_by_price)?;
+    two_order_settlement::settle_two_fillkill_sell_orders(&most_lenient_a, &most_lenient_b)
+}
+
+#[derive(Debug, Default)]
+struct TokenPairOrders {
+    sell_token_0: Vec<UserOrder>,
+    sell_token_1: Vec<UserOrder>,
+}
+
+fn organize_orders_by_token_pair(
+    orders: impl Iterator<Item = UserOrder>,
+) -> HashMap<TokenPair, TokenPairOrders> {
+    let mut result = HashMap::<_, TokenPairOrders>::new();
+    for (order, token_pair) in orders
+        .filter(usable_order)
+        .filter_map(|order| Some((order, order.token_pair()?)))
+    {
+        let token_pair_orders = result.entry(token_pair).or_default();
+        if order.sell_token == token_pair.get().0 {
+            token_pair_orders.sell_token_0.push(order);
+        } else {
+            token_pair_orders.sell_token_1.push(order);
+        }
+    }
+    result
+}
+
+fn usable_order(order: &UserOrder) -> bool {
+    matches!(order.order_kind, OrderKind::Sell)
+        && !order.sell_amount.is_zero()
+        && !order.buy_amount.is_zero()
+        && !order.partially_fillable
```

Comment:
Why do we exclude partially fillable orders? They can also be completely settled. 

Generally, I feel like order filtering (e.g. for 0 amount orders) should happen outside the core solvers (but instead in the order fetching parts of the system).

Reply:
I guess partially fillable would work too but for that we need to fetch the already filled amount first.
With the 0 amount it's not clear whether this could even happen in the orderbook or whether all solvers need this prefiltered. Another solver might have no problem settling orders that have 0 buy amount.